internal/changelog/fragment/template.yaml

```diff
@@ -0,0 +1,26 @@
+# one of:
+# - breaking-change: a change to previously-documented behavior
+# - deprecation: functionality that is being removed in a later release
+# - bug-fix: fixes a problem in a previous version
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: feature
+# Change summary; a 80ish characters long description of the change.
+summary: 
+# Long description; in case the summary is not enough to describe the change this field accomodate a description without length limits.
+# description: 
+# PR number; optional; the PR number that added the changeset.
+# If not present is automatically filled by the tooling finding the PR where this changelog fragment has been added.
+# NOTE: the tooling supports backports, so it's able to fill the original PR number instead of the backport PR number.
+# Please provide it if you are adding a fragment for a different PR.
+# pr: 1234
+# Issue number; optional; the GitHub issue related to this changeset (either closes or is part of).
+# If not present is automatically filled by the tooling with the issue linked to the PR number.
```

Comment:
When looking at the issue, are we able to detect the `closes` or `is part of` relationships? If not, I would not try to fill it automatically as many issues might be linked from a PR.

Reply:
> When looking at the issue, are we able to detect the closes or is part of relationships?

Yes! The linked issue feature works in 2 ways (per [GitHub docs](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)):
- automatically through keywords
- manually through UI

I have a working example of this working. PR #16 references issue #5 (see also `Development` section in right toolbar).

Using this GraphQL query (using the [explorer:](https://docs.github.com/en/graphql/overview/explorer))
```
{
  resource(url: "https://github.com/elastic/elastic-agent-changelog-tool/pull/16") {
    ... on PullRequest {
      closingIssuesReferences(first: 10) {
        nodes {
          number
        }
      }
    }
  }
}
```

this result is returned
```
{
  "data": {
    "resource": {
      "closingIssuesReferences": {
        "nodes": [
          {
            "number": 5
          }
        ]
      }
    }
  }
}
```

I don't think this feature is available using the REST API but it's available through the GraphQL one.